src/modes/intraOrderbook.ts

```diff
@@ -103,13 +108,13 @@ export async function dryrun({
 
     // trying to find opp with doing gas estimation, once to get gas and calculate
     // minimum sender output and second time to check the clear2() with withdraw2() and headroom
-    let gasLimit, blockNumber;
+    let gasLimit, blockNumber, l1Cost;
     try {
         blockNumber = Number(await viemClient.getBlockNumber());
         spanAttributes["blockNumber"] = blockNumber;
-        gasLimit = ethers.BigNumber.from(await signer.estimateGas(rawtx))
-            .mul(config.gasLimitMultiplier)
-            .div(100);
+        const estimation = await estimateGasCost(rawtx, signer, config, l1GasPrice, l1Signer);
+        l1Cost = estimation.l1Cost;
+        gasLimit = ethers.BigNumber.from(estimation.gas).mul(config.gasLimitMultiplier).div(100);
```

Comment:
this is dupe code from interOrderbook?

Reply:
not sure by dup code, it is gascost estimation call and then the result of it is destructured to be used, probably this destructure is present on each mode yes, but not sure if it fits the dup code criteria

Reply:
ofc each mode needs to do a gas estimate call, and use the result downstream of its own logic